src/main/java/com/epam/prejap/teatrees/game/Playfield.java

```diff
@@ -81,12 +76,12 @@ private boolean isValidMove(Block block, int rowOffset, int colOffset) {
     }
 
     private void hide() {
-        forEachBrick((i, j, dot) -> grid[row + i][col + j] = 0);
+        forEachBrick((i, j, dot) -> grid.setCell(row + i, col + j, 0));
     }
 
     private void show() {
-        forEachBrick((i, j, dot) -> grid[row + i][col + j] = dot);
-        printer.draw(grid);
+        forEachBrick((i, j, dot) -> grid.setCell(row + i,col + j, dot));
+        printer.draw(grid.grid());
```

Comment:
Maybe it should be the responsibility of the grid to print itself using the printer?

Reply:
Implemented.